src/documentation/runners/direct.md

```diff
@@ -1,9 +1,47 @@
 ---
 layout: default
-title: "Apache Direct Runner"
+title: "Direct Runner"
 permalink: /documentation/runners/direct/
 redirect_from: /learn/runners/direct/
 ---
 # Using the Direct Runner
 
-This page is under construction ([BEAM-505](https://issues.apache.org/jira/browse/BEAM-505)).
+The direct runner executes pipelines on your machine and is designed to validate that pipelines adhere to the Apache Beam model as closely as possible. Instead of focusing on efficient pipeline execution, the direct runner performs additional checks to ensure that users do not rely on semantics that are not guaranteed by the model. Some of these checks include:
+
+* enforcing immutability of elements
+* enforcing encodability of elements
+* elements are processed in an arbitrary order at all points
+* serialization of user Fns (DoFn, CombineFn, etc.)
+
+Using the direct runner for testing and development helps ensure that pipelines are robust across different Beam runners. In addition, debugging failed runs can be a non-trivial task when a pipeline executes on a remote cluster. Instead, it is often faster and simpler to perform local unit testing on your pipeline code. Unit testing your pipeline locally also allows you to use your preferred local debugging tools.
+
+Here are some resources with information about how to test your pipelines.
+* [Testing Unbounded Pipelines in Apache Beam](https://beam.incubator.apache.org/blog/2016/10/20/test-stream.html) talks about the use of Java classes [PAssert]({{ site.baseurl }}/documentation/sdks/javadoc/{{ site.release_latest }}/index.html?org/apache/beam/sdk/testing/PAssert.html) and [TestStream]({{ site.baseurl }}/documentation/sdks/javadoc/{{ site.release_latest }}/index.html?org/apache/beam/sdk/testing/TestStream.html) to test your pipelines.
+* The [Apache Beam WordCount Example](http://beam.incubator.apache.org/get-started/wordcount-example/) contains an example of logging and testing a pipeline with `PAssert`.
+
+
+## Direct runner prerequisites and setup
+
+You must specify your dependency on the direct runner.
+
+For example, if you are using Maven for development and want to use the SDK for Java with [DirectRunner]({{ site.baseurl }}/documentation/sdks/javadoc/{{ site.release_latest }}/index.html?org/apache/beam/runners/direct/DirectRunner.html), add the following dependency to your `pom.xml` file:
+
+```
+   <dependency>
```

Comment:
Yeah, it's a bit troublesome. If I put the text that specifies it's for Maven/Java outside of the language toggle, it looks strange if the user toggles python... but it also looks strange when I put the text inside the Java toggle because it forces everything inside to be code formatted, along with syntax highlighting. I will see how it looks with the Maven/Java specific text in a comment inside the Java toggle. Other ideas welcome!